integration-tests/src/test/java/com/datastax/oss/driver/api/core/session/ShutdownIT.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.api.core.session;
+
+import static com.datastax.oss.simulacron.common.stubbing.PrimeDsl.noRows;
+import static com.datastax.oss.simulacron.common.stubbing.PrimeDsl.when;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.datastax.oss.driver.api.core.CqlSession;
+import com.datastax.oss.driver.api.core.NoNodeAvailableException;
+import com.datastax.oss.driver.api.core.connection.ClosedConnectionException;
+import com.datastax.oss.driver.api.testinfra.session.SessionUtils;
+import com.datastax.oss.driver.api.testinfra.simulacron.SimulacronRule;
+import com.datastax.oss.driver.categories.ParallelizableTests;
+import com.datastax.oss.simulacron.common.cluster.ClusterSpec;
+import java.util.Set;
+import java.util.concurrent.ConcurrentSkipListSet;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.TimeUnit;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+@Category(ParallelizableTests.class)
+public class ShutdownIT {
+
+  @ClassRule
+  public static SimulacronRule simulacronRule =
+      new SimulacronRule(ClusterSpec.builder().withNodes(1));
+
+  private static final String QUERY_STRING = "select * from foo";
+
+  @Test
+  public void should_fail_requests_when_session_is_closed() throws Exception {
+    // Given
+    // Prime with a bit of delay to increase the chance that a query will be aborted in flight when
+    // we force-close the session
+    simulacronRule
+        .cluster()
+        .prime(when(QUERY_STRING).then(noRows()).delay(20, TimeUnit.MILLISECONDS));
+    CqlSession session = SessionUtils.newSession(simulacronRule);
+
+    // When
+    // Max out the in-flight requests on the connection (from a separate thread pool to get a bit of
+    // contention), then force-close the session abruptly.
+    Set<String> unexpectedErrors = new ConcurrentSkipListSet<>();
+    ExecutorService requestExecutor = Executors.newFixedThreadPool(4);
+    Semaphore semaphore = new Semaphore(1024);
```

Comment:
isn't that `1024` of some configuration on the server side? If it can change it will break this test - maybe we should extract it somehow?
Or parametrize SimulacronRule with `1024` to have that config explicit?

Reply:
It's from `advanced.connection.max-requests-per-connection` in the configuration. I can pull it from there, it will be more explicit.